src/drt/src/db/obj/frBPin.h

```diff
@@ -31,92 +30,40 @@
 
 #include <iostream>
 
-#include "db/obj/frAccess.h"
-#include "db/obj/frBlockObject.h"
-#include "db/obj/frShape.h"
-#include "frBaseTypes.h"
+#include "db/obj/frPin.h"
 
 namespace fr {
-class frTerm;
 class frBTerm;
 
-class frBPin : public frBlockObject
+class frBPin : public frPin
 {
  public:
   // constructors
-  frBPin() : frBlockObject(), term_(nullptr), pinFigs_(), aps_() {}
-  frBPin(const frBPin& in) : frBlockObject(), term_(in.term_)
-  {
-    for (auto& uPinFig : in.getFigs()) {
-      auto pinFig = uPinFig.get();
-      if (pinFig->typeId() == frcRect) {
-        std::unique_ptr<frPinFig> tmp
-            = std::make_unique<frRect>(*static_cast<frRect*>(pinFig));
-        addPinFig(std::move(tmp));
-      } else if (pinFig->typeId() == frcPolygon) {
-        std::unique_ptr<frPinFig> tmp
-            = std::make_unique<frPolygon>(*static_cast<frPolygon*>(pinFig));
-        addPinFig(std::move(tmp));
-      } else {
-        std::cout << "Unsupported pinFig in copy constructor" << std::endl;
-        exit(1);
-      }
-    }
-  }
+  frBPin() : frPin(), term_(nullptr) {}
+  frBPin(const frBPin& in) : frPin(in), term_(in.term_) {}
   frBPin(const frBPin& in, const dbTransform& xform)
-      : frBlockObject(), term_(in.term_)
-  {
-    for (auto& uPinFig : in.getFigs()) {
-      auto pinFig = uPinFig.get();
-      if (pinFig->typeId() == frcRect) {
-        std::unique_ptr<frPinFig> tmp
-            = std::make_unique<frRect>(*static_cast<frRect*>(pinFig));
-        tmp->move(xform);
-        addPinFig(std::move(tmp));
-      } else if (pinFig->typeId() == frcPolygon) {
-        std::unique_ptr<frPinFig> tmp
-            = std::make_unique<frPolygon>(*static_cast<frPolygon*>(pinFig));
-        tmp->move(xform);
-        addPinFig(std::move(tmp));
-      } else {
-        std::cout << "Unsupported pinFig in copy constructor" << std::endl;
-        exit(1);
-      }
-    }
-  }
+      : frPin(in), term_(in.term_) {}
```

Comment:
Why does this now ignore the xform?

Reply:
Good catch, it's supposed to inherit from the `frPin(frBPin&, const dbTransform&)` constructor not the default constructor.